packages/Checkbox/src/Checkbox.tsx

```diff
@@ -160,12 +185,14 @@ export default class Checkbox extends PureComponent {
     // For Microsoft Edge and IE, when checkbox is indeterminate, change event does not fire when clicked.
     // Explicitly call onChange for this case
     if (this.inputRef.current && this.inputRef.current.indeterminate) {
-      this.onChange(e);
+      // Event bubbling means that TS can't make assumptions about the event's target.
```

Comment:
Seems better to override TypeScript than modify the event we're sending to the handler.

Reply:
so `this.onChange(e as unknown as React.ChangeEvent<HTMLInputElement>)`? That doesn't feel better to me, but... 🤷‍♂

Reply:
Can you omit `as unknown`? I think that's uglier but more 'correct' as far as the application code is concerned.

Reply:
without the `as unknown` it complains

> Conversion of type 'MouseEvent<HTMLInputElement, MouseEvent>' to type 'ChangeEvent<HTMLInputElement>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

Reply:
I'm not 100% sure I agree about it being "more correct". Change events are guaranteed to have the input as the event's `target` (according to [this comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12239#issuecomment-256308813) at least), but click events don't have that guarantee.

I think I might've stumbled upon a more technically-correct solution, actually:

```tsx
onClick = (e: React.MouseEvent<HTMLInputElement> & React.ChangeEvent<HTMLInputElement>) => {
  ...
```

Reply:
Or more precisely...

```ts
e: React.MouseEvent<HTMLInputElement> & { target: HTMLInputElement }
```